sycl/doc/MultiTileCardWithLevelZero.md

```diff
@@ -0,0 +1,125 @@
+	
+Contexts are used for resources isolation and sharing. A SYCL context may consist of one or multiple devices.
+Both root-devices and sub-devices can be within single context, but they all should be of the same SYCL platform.
+A SYCL program (kernel_bundle) created against a context with multiple devices will be built to each of the root-devices in the context.
+For context that consists of multiple sub-devices of the same root-device only single build (to that root-device) is needed.
+	
+## 3 Memory
+### 3.1 USM
+
+There are multiple ways to allocate memory:
+
+`malloc_device`:
+- Allocation can only be accessed by the specified device but not by other devices in the context nor by host.
+- The data stays on the device all the time and thus is the fastest available for kernel execution.
+- Explicit copy is needed for transfer data to the host or other devices in the context.
+				   
+`malloc_host`:
+- Allocation can be accessed by the host and any other device in the context.
+- The data stays on the host all the time and is accessed via PCI from the devices.
+- No explicit copy is needed for synchronizing of the data with the host or devices.
+			   
+`malloc_shared`:
+- Allocation can be accessed by the host and the specified device only.
+- The data can migrate (operated by the Level-Zero driver) between the host and the device for faster access.
+- No explicit copy is necessary for synchronizing between the host and the device, but it is needed for other devices in the context.
+				   
+NOTE: Memory allocated against a root-device is accessible by all of its sub-devices (tiles).
+So if operating on a context with multiple sub-devices of the same root-device then you can use `malloc_device` on that root-device instead of using the slower `malloc_host`.
+Remember that if using `malloc_device` you'd need an explicit copy out to the host if it necessary to see data there.
+					   
+### 3.2 Buffer
+	
+SYCL buffers are also created against a context and under the hood are mapped to the Level-Zero USM allocation discussed above.
+The current mapping is following:
+
+- For integrated device the allocations are made on host, and are accessible by the host and the device without any copying.
+- Memory buffers for context with sub-devices of the same root-device (possibly including the root-device itself) are allocated on that root-device.
+   Thus they are readily accessible by all the devices in such context. The synchronization with the host is performed by SYCL RT with map/unmap doing implicit copies when necessary.
+- Memory buffers for context with devices from different root-devices in it are allocated on host (thus made accessible to all devices).
+	
+## 4 Queue
+
+SYCL queue is always attached to a single device in a possibly multi-device context.
+Some typical scenarios are the following (from most performant to least performant):
+
+A. Context with a single sub-device in it and the queue is attached to that sub-device (tile)

```

Comment:
Same here. Might be a good idea to inline short snippets of code with examples.

Reply:
OK, added

Reply:
added